### ISSUE TITLE
Versioning of msvc inconsistent

### ISSUE THREAD

**Issue by terrorfisch** [stefanseefeld/faber#5]
The version of Visual Studio 2017 is 15.x but the version of the toolset is 14.1x
[https://blogs.msdn.microsoft.com/vcblog/2017/11/15/side-by-side-minor-version-msvc-toolsets-in-visual-studio-2017/](url)

The vswhere code uses the toolset version as VS version
https://github.com/stefanseefeld/faber/blob/f1398e5f8d6d447e722dd883334851f6b7ef0fae/src/faber/tools/msvc.py#L198

I am currently working on implementing autodetection of vswhere and stumbled upon this inconsistency. I have VS 2015 and VS 2017 installed and this bug results in overwriting the VS 2017 entry with the VS 2015 entry from the registry.

I guess the correct version would be 14.1 to comply with the boost naming of libraries?

**Comment by stefanseefeld** [comment]
Yeah, I'm a bit stuck in the versioning inconsistency. I'm not sure what the "correct" version is. What version would the original faber code report for VS 2015 and VS 2017 respectively ? And what does it report with your patch ?

**Comment by stefanseefeld** [comment]
Hi @terrorfisch , are you still interested in addressing this issue ? I'm not a regular MSVC user, so can offer only little help. Nevertheless, I'd be interested in fixing any such issue. Thanks !
